Various small improvements

UniqMartin opened the pull request:
*This is a set of various small improvements, best viewed commit by commit. I was not feeling like creating a bunch of different pull requests for these, but could of course do this, if this is preferred.*

Tested on my Mac running OS X 10.10 and everything still worked as it should (or improved, namely plug-in support). I cannot verify these changes on Windows/Linux, but hopefully someone else can.

I'm not entirely sure about commit UniqMartin/texworks@0e5759eaec9d59178da044ecb5b208a24fd69a04. For me, the file `trans/TeXworks_trans.pro` was always regenerated when running `cmake`, but with different contents (files referenced therein in a different order), so I wanted to get rid of this distraction. But maybe its presence in the repository serves a purpose. :)

stloeffler commented:
Most of the commits look good. I'm not sure if using MODULE has any side-effects on Linux or Windows (default seems to be SHARED), I'll have to test before merging when I have time.

`trans/TeXworks_trans.pro` is there for a reason, namely so that people can translate TeXworks without having to build it. Qt's lupdate utility needs to know about all sources, and the easiest way of doing this is to give it a .pro file. Maybe we can simply sort the list of files (in CMake) before generating the file? That way we should have a robust order and the file content should no longer change...

UniqMartin commented:
**Regarding MODULE:** The default is either `STATIC` or `SHARED`, depending on the contents of `BUILD_SHARED_LIBS`. I compared the CMake variables affecting shared library/module compilation by looking at the output of `cmake --system-information`. On Linux (Debian Wheezy, x86_64, CMake 2.8.9) there is no difference at all. On Windows (Windows 7, 32-bit, CMake 3.2.2) the only difference is whether an import library is built or not. Thus I'm pretty confident there won't be side-effects on these platforms. But a proper test would be better, of course. (See [this gist](https://gist.github.com/UniqMartin/62cb51039ca10ebe373d) for details.)

**Regarding `trans/TeXworks_trans.pro`:** Thanks for confirming my suspicion! I replaced my original commit with one that sorts the list of files before writing them to `trans/TeXworks_trans.pro`, as you suggested. I also updated the file with the pre-sorted generated version.

**One more thing:** I also noticed, that while being obsoleted by CMake, there still existed project files for `qmake`. I removed them in an additional commit.

stloeffler commented:
Thanks for the work - merged.
